packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.kt

```diff
@@ -612,11 +614,25 @@ class UpdatesController private constructor(
      */
     @JvmStatic fun initialize(context: Context, configuration: Map<String, Any>) {
       if (singletonInstance == null) {
-        val updatesConfiguration = UpdatesConfiguration(context, configuration)
+        val updatesConfiguration = _overrideConfiguration ?: UpdatesConfiguration(context, configuration)
         singletonInstance = UpdatesController(context, updatesConfiguration)
         singletonInstance!!.start(context)
       }
     }
+
+    /**
+     * Overrides the [UpdatesConfiguration] that will be used inside [UpdatesController]
+     */
+    @JvmStatic
+    fun overrideConfiguration(context: Context, configuration: Map<String, Any>) {
+      if (Looper.myLooper() != Looper.getMainLooper()) {
+        throw AssertionError("overrideConfiguration() should be called from main thread")
+      }
```

Comment:
I think it should theoretically be fine to call this from another thread as long as the `singletonInstance != null` check below holds.